crates/router/src/core/payments/helpers.rs

```diff
@@ -2517,22 +2527,40 @@ pub async fn get_additional_payment_data(
                     .map_err(|error| services::logger::warn!(card_info_error=?error))
                     .ok()
                     .flatten()
-                    .map(
-                        |card_info| api_models::payments::AdditionalPaymentData::Card {
-                            card_issuer: card_info.card_issuer,
-                            card_network: card_info.card_network.clone(),
-                            bank_code: card_info.bank_code,
-                            card_type: card_info.card_type,
-                            card_issuing_country: card_info.card_issuing_country,
-                        },
-                    );
-                card_info.unwrap_or(api_models::payments::AdditionalPaymentData::Card {
-                    card_issuer: None,
-                    card_network: None,
-                    bank_code: None,
-                    card_type: None,
-                    card_issuing_country: None,
-                })
+                    .map(|card_info| {
+                        api_models::payments::AdditionalPaymentData::Card(Box::new(
+                            api_models::payments::AdditionalCardInfo {
+                                card_issuer: card_info.card_issuer,
+                                card_network: card_info.card_network.clone(),
+                                bank_code: card_info.bank_code,
+                                card_type: card_info.card_type,
+                                card_issuing_country: card_info.card_issuing_country,
+                                last4: card_data.card_number.peek().clone()
+                                    [card_number_length - 4..card_number_length]
+                                    .to_string(),
+                                card_isin: card_data.card_number.peek().clone()[..6].to_string(),
```

Comment:
Avoid indexing as it can cause panics, incorporate the changes suggested by Narayan.

Reply:
done